ratatui-core/src/backend/test.rs

```diff
@@ -67,6 +68,225 @@ fn buffer_view(buffer: &Buffer) -> String {
     view
 }
 
+/// Returns a string representation of the given buffer including style information for debugging
+/// purpose.
+///
+/// This function is used to visualize the buffer content with style information in a human-readable
+/// format. Style information is represented as `{style}text{/style}` where style can be color
+/// names, modifiers, or combinations thereof.
+fn buffer_view_with_style(buffer: &Buffer) -> String {
+    let mut view =
+        String::with_capacity(buffer.content.len() * 2 + buffer.area.height as usize * 3);
+    for cells in buffer.content.chunks(buffer.area.width as usize) {
+        let mut overwritten = vec![];
+        let mut skip: usize = 0;
+        let mut current_style = crate::style::Style::default();
+        view.push('"');
+        for (x, c) in cells.iter().enumerate() {
+            if skip == 0 {
+                // Only output style tags when the style changes
+                if c.style() != current_style {
+                    // Close previous style if needed
+                    if current_style != crate::style::Style::default() {
+                        let style_end = format_style_end(&current_style);
+                        if !style_end.is_empty() {
+                            view.push_str(&style_end);
+                        }
+                    }
+                    // Open new style if needed
+                    if c.style() != crate::style::Style::default() {
+                        let style_start = format_style_start(&c.style());
+                        if !style_start.is_empty() {
+                            view.push_str(&style_start);
+                        }
+                    }
+                    current_style = c.style();
+                }
+                view.push_str(c.symbol());
+            } else {
+                overwritten.push((x, c.symbol()));
+            }
+            skip = core::cmp::max(skip, c.symbol().width()).saturating_sub(1);
+        }
+        // Close any remaining style tags at the end of the line
+        if current_style != crate::style::Style::default() {
+            let style_end = format_style_end(&current_style);
+            if !style_end.is_empty() {
+                view.push_str(&style_end);
+            }
+        }
+        view.push('"');
+        if !overwritten.is_empty() {
+            write!(&mut view, " Hidden by multi-width symbols: {overwritten:?}").unwrap();
+        }
```

Comment:
Not sure if this is the best representation for the multi-width symbols. How does this exactly look? We should add tests for it and avoid panicking via `unwrap()`.